rustls/src/key_log_file.rs

```diff
@@ -1,6 +1,7 @@
 #[cfg(feature = "logging")]
 use crate::log::warn;
 use crate::KeyLog;
+use alloc::{string::String, vec::Vec};
```

Comment:
Could you maintain the whitespace between the `use crate::*` imports and the std /core imports here please :bow:

Reply:
Also I think in most modules we keep core/std imports below crate imports, which we should either globally fix or adhere to here.

Reply:
>  we keep core/std imports below crate imports

where do `self` and `super` imports fall in that picture?

AFAICT, neither rust-analyzer or rustfmt allows grouping imports in a way that `crate` imports precede imports from the `std` facade. [rust-analyzer uses this ordering](https://rust-analyzer.github.io/manual.html#auto-import): 

- `std` and `core`
- External Crates
- Current Crate, paths prefixed by `crate`
- Current Module, paths prefixed by `self`
- Super Module, paths prefixed by `super`

and you can get a similar ordering using this unstable rustfmt setting:

``` toml
group_imports = "StdExternalCrate"
```

except that `crate`, `self` and `super` imports are lumped together.

Reply:
I would be in favor of a commit or PR that reformats all modules in the repo according to the StdExternalCrate order.

Reply:
there's a rust-analyzer feature request to allow custom ordering of those 5 import groups: https://github.com/rust-lang/rust-analyzer/issues/13960